resolver/addressable_resolver.go

```diff
@@ -63,50 +63,31 @@ func NewURIResolver(ctx context.Context, callback func(types.NamespacedName)) *U
 }
 
 // URIFromDestination resolves a Destination into a URI string.
-func (r *URIResolver) URIFromDestination(dest apisv1alpha1.Destination, parent interface{}) (string, error) {
-	var deprecatedObjectReference *corev1.ObjectReference
-	if dest.DeprecatedAPIVersion == "" && dest.DeprecatedKind == "" && dest.DeprecatedName == "" && dest.DeprecatedNamespace == "" {
-		deprecatedObjectReference = nil
-	} else {
-		deprecatedObjectReference = &corev1.ObjectReference{
-			Kind:       dest.DeprecatedKind,
-			APIVersion: dest.DeprecatedAPIVersion,
-			Name:       dest.DeprecatedName,
-			Namespace:  dest.DeprecatedNamespace,
-		}
-	}
-	if dest.Ref != nil && deprecatedObjectReference != nil {
-		return "", fmt.Errorf("ref and [apiVersion, kind, name] can't be both present")
-	}
-	var ref *corev1.ObjectReference
-	if dest.Ref != nil {
-		ref = dest.Ref
-	} else {
-		ref = deprecatedObjectReference
+func (r *URIResolver) URIFromDestination(ctx context.Context, namespace string, dest apisv1alpha1.Destination, parent interface{}) (string, error) {
+	err := dest.Validate(ctx)
+	if err != nil {
+		return "", errors.New(err.Error())
 	}
+	ref := dest.GetRef()
 	if ref != nil {
+		// Do not modify original copy w/o namespace.
+		ref = ref.DeepCopy()
+		ref.Namespace = namespace
 		url, err := r.URIFromObjectReference(ref, parent)
 		if err != nil {
 			return "", err
 		}
 		if dest.URI != nil {
-			if dest.URI.URL().IsAbs() {
-				return "", fmt.Errorf("absolute URI is not allowed when Ref or [apiVersion, kind, name] exists")
```

Comment:
Is this no longer true? I think without this check, it means that if `dest.URI` is an absolute URI, then `dest.Ref` is ignored entirely. Is that a correct interpretation?

Reply:
I removed it as I think is already taken care in the validate of the destination done above